src/generic_chunk.h

```diff
@@ -25,11 +25,26 @@ typedef enum {
     CHUNK_COMPRESSED 
 } CHUNK_TYPES_T;
 
+typedef struct AddCtx {
+    //RedisModuleDict *dict;
+    Chunk_t *inChunk;       // original chunk
+    Chunk_t *outChunk_1;    // used if 1st sample changed
+    Chunk_t *outChunk_2;    // used if chunk was splitted
+    
+    Sample sample;
+    
+    int sz;                 // change in chunk size
+    UpsertType type;
+    short maxSamples;       // used for split
+    bool reindex;
```

Comment:
This should be contained in the Series level, the notion of what is being indexed is contained outside of the chunk.

Reply:
The series is a permanent struct, here we need a flag for this run-time operation